src/main/java/org/testng/TestNG.java

```diff
@@ -1530,14 +1530,9 @@ public void configure(Map cmdLineArgs) {
       result.configFailurePolicy = failurePolicy;
     }
 
```

Comment:
It won't work with Integer which may be provided by 3rd party integration.

Reply:
I didn't quite understand what that means. I have tried this locally with Maven and it works. All we are doing here is to accept the parameter as a String and then parse it. Can you please tell me in which circumstances and with which 3rd party integrations it wouldn't work ?

Reply:
Did you test old surefire versions? Does it work with the command line or ant integration?
What about IDE plugins?
Just be sure the Integer version is not used somewhere.

Reply:
@juherr - I have only tested this with surefire plugin.  So going by what you are saying there can be just two possibilities. 

1. The 3rd party integration provided us with an `Integer` (And we attempted at parsing it as `String` thus causing a `ClassCastException`
2. The 3rd party integration provided us with a `String` and our implementation works.

I guess I will go back and change this to run an `instanceof` check and do it accordingly. 

Have amended my commit and pushed in the changes for this. Please help take a look.